src/io/linux.zig

```diff
@@ -398,6 +403,25 @@ pub const IO = struct {
         };
     }
 
+    fn awaiting_verify(self: *const IO) void {
+        var awaiting_count: u32 = 0;
+        var awaiting = self.awaiting;
+
+        if (awaiting) |completion| {
+            assert(completion.awaiting_previous == null);
+        }
+
+        while (awaiting) |completion| {
+            const awaiting_next = completion.awaiting_next;
+            if (awaiting_next) |next| {
```

Comment:
Future note: would be nice to share this doubly-linked-list logic in fifo.zig or similar for `io/darwin.zig` and `io/windows.zig`